lisa/tests/cpufreq/sanity.py

```diff
@@ -14,51 +14,81 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import os
 
 from lisa.tests.base import Result, ResultBundle, TestBundle
 from lisa.wlgen.sysbench import Sysbench
 from lisa.target import Target
-from lisa.utils import ArtifactPath
+from lisa.utils import ArtifactPath, groupby
+from lisa.analysis.tasks import TasksAnalysis
+
+class UserspaceSanityItem(TestBundle):
+    """
+    Record the number of sysbench events on a given CPU at a given frequency.
+    """
+
+    def __init__(self, res_dir, plat_info, cpu, freq, work):
+        super().__init__(res_dir, plat_info)
+
+        self.cpu = cpu
+        self.freq = freq
+        self.work = work
+
+    @classmethod
+    def _from_target(cls, target, res_dir, cpu, freq):
+        sysbench = Sysbench(target, "sysbench", res_dir)
+        work = None
+
+        with target.cpufreq.use_governor("userspace"):
```

Comment:
We used to have one governor change per freq domain, now it's one per frequency - that's going to considerably slow down the execution of this test. We could either move the governor (and frequency?) change to `UserspaceSanity`, or have both options and add a boolean parameter in to `UserspaceSanityItem._from_target()` to control whether the governor switch is done by the caller.

Reply:
It does double the execution time on Pixel 3. Just moving the change to the other level is not really a good idea, as it would prevent correct operations when creating an item for a given CPU directly, which is the whole point of this PR (would need some annotations for that, I'll add them after #1018 is merged).

A third option that does not require making the API more clunky is to just wrap the target with a cpufreq function not switching a governor to the same.